name.abuchen.portfolio.tests/src/name/abuchen/portfolio/util/TradeCalendarTest.java

```diff
@@ -310,6 +416,241 @@ public void testTSX()
+        assertThat(calendar.isHoliday(LocalDate.parse("2023-04-07")), is(true));
+        
+        // Eastern
+        assertThat(calendar.isHoliday(LocalDate.parse("2023-04-07")), is(true));
+        assertThat(calendar.isHoliday(LocalDate.parse("2023-04-10")), is(true));
+
+        // Labour day
+        assertThat(calendar.isHoliday(LocalDate.parse("2019-05-01")), is(true));
+        assertThat(calendar.isHoliday(LocalDate.parse("2020-05-01")), is(true));
+        assertThat(calendar.isHoliday(LocalDate.parse("2021-05-01")), is(true));
+        assertThat(calendar.isHoliday(LocalDate.parse("2022-05-01")), is(true));
+        assertThat(calendar.isHoliday(LocalDate.parse("2023-05-01")), is(true));
+        
+        // Assumption day
+        assertThat(calendar.isHoliday(LocalDate.parse("2023-08-15")), is(true));
+
+        // Chrismas
+        assertThat(calendar.isHoliday(LocalDate.parse("2023-12-24")), is(true));
+        assertThat(calendar.isHoliday(LocalDate.parse("2023-12-25")), is(true));
+        assertThat(calendar.isHoliday(LocalDate.parse("2023-12-26")), is(true));
+    }
+
+    @Test
+    public void testTradeCalenderVSE()
+    {
+        // See https://www.wienerborse.at/handel/handelsinformationen/handelskalender/
+
+        TradeCalendar calendar = TradeCalendarManager.getInstance("vse");
+
+        // New Year
+        assertThat(calendar.isHoliday(LocalDate.parse("2022-12-31")), is(true));
+        assertThat(calendar.isHoliday(LocalDate.parse("2023-01-01")), is(true));
+
+        // Eastern
+        assertThat(calendar.isHoliday(LocalDate.parse("2023-04-07")), is(true));
+        assertThat(calendar.isHoliday(LocalDate.parse("2023-04-10")), is(true));
+
+        // Whit Monday
+        assertThat(calendar.isHoliday(LocalDate.parse("2021-05-24")), is(true));
+        assertThat(calendar.isHoliday(LocalDate.parse("2022-06-06")), is(true));
+        assertThat(calendar.isHoliday(LocalDate.parse("2023-05-30")), is(false));
+
+        // Labour day
+        assertThat(calendar.isHoliday(LocalDate.parse("2019-05-01")), is(true));
+        assertThat(calendar.isHoliday(LocalDate.parse("2020-05-01")), is(true));
+        assertThat(calendar.isHoliday(LocalDate.parse("2021-05-01")), is(true));
+        assertThat(calendar.isHoliday(LocalDate.parse("2022-05-01")), is(true));
+        assertThat(calendar.isHoliday(LocalDate.parse("2023-05-01")), is(true));
+        
+        // National day
+        assertThat(calendar.isHoliday(LocalDate.parse("2023-10-26")), is(true));
+
+        // Chrismas
+        assertThat(calendar.isHoliday(LocalDate.parse("2023-12-24")), is(true));
+        assertThat(calendar.isHoliday(LocalDate.parse("2023-12-25")), is(true));
+
+        // Saint Stephen
+        assertThat(calendar.isHoliday(LocalDate.parse("2023-12-26")), is(true));
+    }
+
+    @Test
+    public void testTradeCalenderMICEXRTS()
+    {
+        // See https://www.wienerborse.at/handel/handelsinformationen/handelskalender/
```

Comment:
Certainly not the place to look for information about Moscow stock exchange.

Reply:
thats right... my fault.